nilearn/plotting/surface/_utils.py

```diff
@@ -86,42 +94,111 @@ def _check_bg_map(bg_map, hemi):
+        if hemi in surf_mesh.parts:
+            return surf_mesh.parts[hemi]
+        else:
+            raise ValueError(
+                f"{hemi=} does not exist in mesh. Available hemispheres are:"
+                f"{surf_mesh.parts.keys()}."
+            )
     else:
-        raise ValueError("hemi must be one of left, right or both.")
+        raise ValueError("hemi must be one of 'left', 'right' or 'both'.")
 
 
+@fill_doc
 def check_surface_plotting_inputs(
     surf_map,
     surf_mesh,
-    hemi="left",
+    hemi=DEFAULT_HEMI,
     bg_map=None,
     map_var_name="surf_map",
     mesh_var_name="surf_mesh",
 ):
     """Check inputs for surface plotting.
 
-    Where possible this will 'convert' the inputs
-    if SurfaceImage or PolyMesh objects are passed
-    to be able to give them to the surface plotting functions.
+    Where possible this will 'convert' the inputs if
+    :obj:`~nilearn.surface.SurfaceImage` or :obj:`~nilearn.surface.PolyMesh`
+    objects are passed to be able to give them to the surface plotting
+    functions.
+
+    - ``surf_mesh`` and ``surf_map`` cannot be `None` at the same time.
+    - If ``surf_mesh=None``, then ``surf_map`` should be of type
+    :obj:`~nilearn.surface.SurfaceImage`.
+    - ``surf_mesh`` cannot be of type :obj:`~nilearn.surface.SurfaceImage`.
+    - If ``surf_map`` and ``bg_map`` are of type
+    :obj:`~nilearn.surface.SurfaceImage`, ``bg_map.mesh`` should be equal to
+    ``surf_map.mesh``.
+
+    Parameters
+    ----------
+    surf_map: :obj:`~nilearn.surface.SurfaceImage` | :obj:`numpy.ndarray`
+              | None
+
+    %(surf_mesh)s
+        If `None` is passed, then ``surf_map`` must be a
+        :obj:`~nilearn.surface.SurfaceImage` instance and the mesh from that
+        :obj:`~nilearn.surface.SurfaceImage` instance will be used.
+
+    %(hemi)s
+
+    %(bg_map)s
 
     Returns
     -------
-    surf_map : numpy.ndarray
+    surf_map : :obj:`numpy.ndarray`
 
-    surf_mesh : numpy.ndarray
+    surf_mesh : :obj:`numpy.ndarray`,  :obj:`~nilearn.surface.InMemoryMesh`
+        Surface mesh corresponding to the specified ``hemi``.
 
-    bg_map : str | pathlib.Path | numpy.ndarray | None
+        - If ``hemi='left'`` or ``hemi='right'``, returns
+          :obj:`numpy.ndarray`.
+        - If ``hemi='both'``, returns :obj:`~nilearn.surface.InMemoryMesh`
+    bg_map : :obj:`str` | :obj:`pathlib.Path` | :obj:`numpy.ndarray` | None
 
     """
+    if not isinstance(surf_map, SurfaceImage) and not isinstance(
+        surf_mesh, PolyMesh
+    ):
+        warn(
+            category=UserWarning,
+            message=(
+                f"{hemi=} was passed with "
+                f"{type(surf_map)=} and {type(surf_mesh)=}.\n"
+                "This value will be ignored as it is only used when "
+                "'map' is a SurfaceImage instance and / or "
+                "'mesh' is a PolyMesh instance."
+            ),
+            stacklevel=3,
+        )
+
```

Comment:
I moved the warning for `sanitize_hemi_...` here in `check_surface_plotting_inputs`.